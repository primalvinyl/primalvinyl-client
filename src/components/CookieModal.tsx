import React from 'react';
import CookieConsent from 'react-cookie-consent';
import { Link } from 'react-router-dom';
import './CookieModal.scss';

const CookieConsentModal = (): React.ReactElement => {
    return (
        <div className="root">
            <CookieConsent
                containerClasses="cookieConsent"
                contentClasses="contentWrapper"
                buttonWrapperClasses="buttonWrapper"
                location="bottom"
                buttonText="Accept"
                cookieName="botterppotterconsent"
                overlay={true}
                expires={150}>
                <h3>
                    We Use Cookies
                </h3>
                <p> 
                    By clicking the Accept button, you agree to our and our parter's use of cookies and other technologies to process your data to personalise your experience and deliver tailored advertising to you, on and off our site.
                </p>
                <p>
                    To find out more, please read our <Link to="/terms-of-service">privacy policy</Link>.
                </p>
            </CookieConsent>
        </div>
    );
};

export default CookieConsentModal;
