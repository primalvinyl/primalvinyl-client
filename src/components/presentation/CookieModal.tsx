import React from 'react';
import CookieConsent from 'react-cookie-consent';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const Wrapper = styled.div`
    .cookieConsent {
        .contentWrapper {
            padding: .5em 1em 1em 1em;
            font-weight: normal;

            p {
                margin: 0 0 .8em 0;
            }

            h3 {
                font-size: 1.8em;
            }
        }
        .buttonWrapper {
            padding: 0 2em 1em 1em;
        }
    }
`;

const CookieModal = ({ expires }: CookieModalProps): React.ReactElement => {
    return (
        <Wrapper>
            <CookieConsent
                containerClasses="cookieConsent"
                contentClasses="contentWrapper"
                buttonWrapperClasses="buttonWrapper"
                location="bottom"
                buttonText="Accept"
                cookieName="trackingconsent"
                overlay={true}
                expires={expires}>
                <h3>
                    We Use Cookies
                </h3>
                <p>
                    By clicking the Accept button, you agree to our and our parter's use of cookies and other technologies to process your data to personalise your experience and deliver tailored advertising to you, on and off our site.
                </p>
                <p>
                    To find out more, please read our <Link to="/terms-of-service">privacy policy</Link>.
                </p>
            </CookieConsent>
        </Wrapper>
    );
};

type CookieModalProps = {
    readonly expires?: number;
};

CookieModal.defaultProps = {
    expires: 150
};

export default CookieModal;
