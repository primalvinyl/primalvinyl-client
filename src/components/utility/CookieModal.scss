@import '../styles/partials/breakpoints';
@import '../styles/partials/variables';

.root {
    .cookieConsent {
        .contentWrapper {
            padding: .5em 1em 1em 1em;
            font-weight: normal;

            h3 {
                font-size: 1.8em;
            }
        }
        .buttonWrapper {
            padding: 0 2em 1em 1em;
        }

        @include breakpoint(lg-down){
            .contentWrapper {
                padding: .5em 1em 0 1em;
        
                p {
                    margin: 0 0 .8em 0;
                }
            }
            .buttonWrapper {
                padding: 0 1em 1.5em 1em;
            }
        }
    }
}
