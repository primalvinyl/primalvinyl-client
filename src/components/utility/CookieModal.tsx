import React from 'react';
import CookieConsent from 'react-cookie-consent';
import { Link } from 'react-router-dom';
import styles from './CookieModal.module.scss';

const CookieModal = (): React.ReactElement => {
    return (
        <div className={styles.root}>
            <CookieConsent
                containerClasses={styles.cookieConsent}
                contentClasses={styles.contentWrapper}
                buttonWrapperClasses={styles.buttonWrapper}
                location="bottom"
                buttonText="Accept"
                cookieName="trackingconsent"
                overlay={true}
                expires={150}>
                <h3>
                    We Use Cookies
                </h3>
                <p> 
                    By clicking the Accept button, you agree to our and our parter's use of cookies and other technologies to process your data to personalise your experience and deliver tailored advertising to you, on and off our site.
                </p>
                <p>
                    To find out more, please read our <Link to="/terms-of-service">privacy policy</Link>.
                </p>
            </CookieConsent>
        </div>
    );
};

export default CookieModal;
