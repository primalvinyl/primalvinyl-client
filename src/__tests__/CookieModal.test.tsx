import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import CookieModal from '../components/CookieModal';

describe('CookieModal', () => {
    it('renders', () => {
        const myWrapper = shallow(
            <MemoryRouter>
                <CookieModal />
            </MemoryRouter>
        );
        expect(myWrapper.exists());
    });

    it('matches snapshot without props', () => {
        const component = Renderer.create(
            <MemoryRouter>
                <CookieModal />
            </MemoryRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
