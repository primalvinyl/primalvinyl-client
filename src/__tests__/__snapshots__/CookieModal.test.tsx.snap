// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieModal matches snapshot without props 1`] = `
<div
  className="sc-bdfBwQ hVwrW"
>
  <div
    className=""
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
        "height": "100%",
        "left": 0,
        "position": "fixed",
        "top": 0,
        "width": "100%",
        "zIndex": "999",
      }
    }
  >
    <div
      className="cookieConsent"
      style={
        Object {
          "alignItems": "baseline",
          "background": "#353535",
          "bottom": "0",
          "color": "white",
          "display": "flex",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "left": "0",
          "position": "fixed",
          "width": "100%",
          "zIndex": "999",
        }
      }
    >
      <div
        className="contentWrapper"
        style={
          Object {
            "flex": "1 0 300px",
            "margin": "15px",
          }
        }
      >
        <h3>
          We Use Cookies
        </h3>
        <p>
          By clicking the Accept button, you agree to our and our parter's use of cookies and other technologies to process your data to personalise your experience and deliver tailored advertising to you, on and off our site.
        </p>
        <p>
          To find out more, please read our 
          <a
            href="/terms-of-service"
            onClick={[Function]}
          >
            privacy policy
          </a>
          .
        </p>
      </div>
      <div
        className="buttonWrapper"
      >
        <button
          aria-label="Accept cookies"
          className=""
          id="rcc-confirm-button"
          onClick={[Function]}
          style={
            Object {
              "background": "#ffd42d",
              "border": "0",
              "borderRadius": "0px",
              "boxShadow": "none",
              "color": "black",
              "cursor": "pointer",
              "flex": "0 0 auto",
              "margin": "15px",
              "padding": "5px 10px",
            }
          }
        >
          Accept
        </button>
      </div>
    </div>
  </div>
</div>
`;
